### ISSUE TITLE
Cannot convert undefined or null to object on watcher

### ISSUE THREAD

**Issue by Slyke** [s-panferov/awesome-typescript-loader#378]
This has only started happening since yesterday.

Webpack compiles successfully, but if I change a file, the script crashes at this location. 

I'm using webpack-dev-server ^2.2.0 and awesome-typescript-loader ^3.0.0-beta.18

```
PROJECTPATH\node_modules\awesome-typescript-loader\src\instance.ts:334
        const changedFiles = Object.keys(mtimes).map(toUnix);
                                    ^
TypeError: Cannot convert undefined or null to object
    at Compiler.<anonymous> (PROJECTPATH\node_modules\awesome-typescript-loader\src\instance.ts:334:37)
    at next (PROJECTPATH\node_modules\tapable\lib\Tapable.js:140:14)
    at Compiler.compilerInvalid (PROJECTPATH\node_modules\webpack-dev-middleware\lib\Shared.js:144:5)
    at next (PROJECTPATH\node_modules\tapable\lib\Tapable.js:140:14)
    at Compiler.compiler.plugin (PROJECTPATH\node_modules\webpack\lib\CachePlugin.js:32:5)
    at Compiler.applyPluginsAsyncSeries (PROJECTPATH\node_modules\tapable\lib\Tapable.js:142:13)
    at Watching._go (PROJECTPATH\node_modules\webpack\lib\Compiler.js:43:16)
    at Watching.invalidate (PROJECTPATH\node_modules\webpack\lib\Compiler.js:121:8)
    at Watching.<anonymous> (PROJECTPATH\node_modules\webpack\lib\Compiler.js:106:8)
    at EventEmitter.watcher.once (PROJECTPATH\node_modules\webpack\lib\node\NodeWatchFileSystem.js:44:4)
    at EventEmitter.g (events.js:291:16)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at EventEmitter._onTimeout (PROJECTPATH\node_modules\watchpack\lib\watchpack.js:139:7)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
```

**Comment by sshev** [comment]
@s-panferov I'm having the same issue
looks like webpack's dependency `watchpack` updated to it and no longer returns `mtimes` (see here https://github.com/webpack/watchpack/commit/2ed42ccc44d501c22933f46d7f746600b6e45085#diff-ac5cbed7c4dfab9eb42cd4836f36ea7bL19)
therefor `Object.keys(mtimes)` fails

**Comment by Mc-Mido** [comment]
@sshev Thanks, it seems this is since the watchpack 1.3.0 version.

Workaround:
`
npm shrinkwrap --dev
`

Edit the npmshrinkwrap.json and change the watchpack version:
```
"watchpack" : {
                    "version" : "1.2.1",
                    "from" : "watchpack@>=1.2.0 <2.0.0",
                    "resolved" : "http://.../watchpack/-/watchpack-1.2.1.tgz"
}
```
Then npm clean and install

**Comment by ZuBB** [comment]
do no do anything. they shipped new version (1.3.1) where they readded `mtime` obj
